server/graphql/v2/mutation/CreateCollectiveMutation.js

```diff
@@ -22,96 +24,119 @@ async function createCollective(_, args, req) {
 
   const { remoteUser, loaders } = req;
 
-  if (!remoteUser) {
-    throw new Unauthorized('You need to be logged in to create a collective');
-  }
-
-  const collectiveData = {
-    slug: args.collective.slug.toLowerCase(),
-    ...pick(args.collective, ['name', 'description', 'tags']),
-    isActive: false,
-    CreatedByUserId: remoteUser.id,
-    settings: { ...DEFAULT_COLLECTIVE_SETTINGS, ...args.collective.settings },
-  };
-
-  if (isCollectiveSlugReserved(collectiveData.slug)) {
-    throw new Error(`The slug '${collectiveData.slug}' is not allowed.`);
-  }
-  const collectiveWithSlug = await models.Collective.findOne({ where: { slug: collectiveData.slug } });
-  if (collectiveWithSlug) {
-    throw new Error(`The slug ${collectiveData.slug} is already taken. Please use another slug for your collective.`);
-  }
-
-  let host;
-
-  // Handle GitHub automated approval and apply to the Open Source Collective Host
-  if (args.automateApprovalWithGithub && args.collective.githubHandle) {
-    const githubHandle = args.collective.githubHandle;
-    const opensourceHost = defaultHostCollective('opensource');
-    host = await loaders.Collective.byId.load(opensourceHost.CollectiveId);
-    try {
-      const githubAccount = await models.ConnectedAccount.findOne({
-        where: { CollectiveId: remoteUser.CollectiveId, service: 'github' },
-      });
-      if (!githubAccount) {
-        throw new Error('You must have a connected GitHub Account to create a collective with GitHub.');
+  let user = remoteUser,
+    host;
+
+  return sequelize
+    .transaction(async transaction => {
+      if (!user && args.user && args.host.legacyId === defaultHostCollective('foundation').CollectiveId) {
```

Comment:
We should not check against `args.host.legacyId`, because it's a valid usage to specify the host by `id` rather than `legacyId` (ie. `host: { id: 'xxxxx-yyyyy' }`). We should fetch the host from the DB `fetchAccountWithReference` (that can be done outside of the transaction) and the check should be:

```suggestion
      if (!user && args.user && host.id === defaultHostCollective('foundation').CollectiveId) {
```